website/server/controllers/top-level/dataexport.js

```diff
@@ -243,4 +243,49 @@ api.exportUserAvatarPng = {
   },
 };
 
+/**
+ * @api {get} /export/inbox.html Export user private messages as HTML document
+ * @apiVersion 3.0.0
+ * @apiName ExportUserPrivateMessages
+ * @apiGroup DataExport
+ * @apiDescription NOTE: Part of the private API that may change at any time.
+ *
+ * @apiSuccess {String} An html page
+ */
+api.exportUserPrivateMessages = {
+  method: 'GET',
+  url: '/export/inbox.html',
+  middlewares: [authWithSession],
+  async handler (req, res) {
+    const timezoneOffset = res.locals.user.preferences.timezoneOffset;
+    const dateFormat = res.locals.user.preferences.dateFormat.toUpperCase();
+
+    let inbox = Object.keys(res.locals.user.inbox.messages).map(key => res.locals.user.inbox.messages[key]);
+
+    // Sort array in descending order
+    inbox = _.sortBy(inbox, function sortBy (num) {
+      return num.sort * -1;
+    });
+
+    let messages = '<!DOCTYPE html><html><head></head><body>';
+
+    let recipient;
+
+    for (let i = 0; i < inbox.length; i++) {
+      recipient = inbox[i].sent ? '=>' : '<=';
+      messages += `<p>${recipient}<strong>${inbox[i].user}</strong> ${moment.utc(inbox[i].timestamp).zone(timezoneOffset).format(`${dateFormat } HH:mm:ss`)} 
+        (${i + 1}/${inbox.length})<br />${inbox[i].text}</p>`;
```

Comment:
@Althenis When this PR (https://github.com/HabitRPG/habitrpg/pull/8030) is merged in, you'll be able to do this to render the markdown as html:

```js
import md from 'habitica-markdown';

let text = md.render(inbox[i].text);
```